src/lib/FeedbackForm.svelte

```diff
@@ -1,53 +1,153 @@
 <script lang="ts">
 
+  import FormMessage from './FormMessage.svelte';
+  import { slide } from 'svelte/transition';
   import '@shoelace-style/shoelace/dist/components/button/button.js';
   import '@shoelace-style/shoelace/dist/components/input/input.js';
   import '@shoelace-style/shoelace/dist/components/textarea/textarea.js';
   import { serialize } from '@shoelace-style/shoelace/dist/utilities/form.js'
+  import '@shoelace-style/shoelace/dist/components/alert/alert.js';
    
   let userURL: string = location.href;
   let userAgent: string = navigator.userAgent;
-  
-  const onSubmit = async () => {
+
+  //takes long string output of document.cookie and splits it into a usable javascript object
+  let cookies: object = document.cookie
+  .split(';')
+  .map(cookie => cookie.split('='))
+  .reduce((accumulator, [key, value]) => ({ ...accumulator, [key.trim()]: decodeURIComponent(value) }), {});
+
+  //if user isn't logged in, HTstatus cookie won't exist
+  let userAuthStatus: string = cookies.HTstatus || 'not logged in';
+
+  let postResponseStatusCode;
+
+  // when true, spinner on submit button animates
+  let loading = false;
+  // when true, hides the element (in this case, the form)
+  let hidden = false;
+  // when true, shows the success/failure alert message
+  let submitted = false;
+
+  const postForm = async () => {
     //shoelace serializer for turning FormData into JSON
     const form = document.querySelector('form')
     const data = serialize(form)
 
-    const res = await fetch('http://localhost:5000/api', {
-          method: 'POST', 
-            body: JSON.stringify(data),
-            headers: {
-                'Content-Type': 'application/json',
-                'Accept': 'application/json'
-            }
-        })
-
-        const json = await res.json()
-        console.log(JSON.stringify(json))
+    const response = await fetch('http://localhost:5000/api', {
+        method: 'POST', 
+          body: JSON.stringify(data),
+          headers: {
+              'Content-Type': 'application/json',
+              'Accept': 'application/json'
+          }
+    })
+        
+    // did something go wrong with the fetch?
+    // if yes, the function stops here
+    if (!response.ok) {
+      postResponseStatusCode = response.status; 
+      throw new Error(`status: ${response.status}`);
     }
+
+    // otherwise, return the jira response json data
+    const json = await response.json()
+    postResponseStatusCode = response.status;
+
+    return json;
+  }
+
+  // handles front-end reaction to form submission
+  const onSubmit = () => {
+      // set the submit button spinner spinning
+      loading = true;
+
+      // do the post fetch function
+      postForm()
+      
+      // if no error, hide form and log new issue ID
+      .then(jiraResponseData => {
```

Comment:
Do you get back a promise from `postForm()` using the `await`? Maybe 

```
return fetch('http://localhost:5000/api')
.then(response => {
  if (!response.ok) {
    // throw exception
  }
  postResponseStatusCode = response.status;
  return response.json()
})
```

...and then you'd get back a promise in `onSubmit`

Reply:
All fixed! Dumb promises 🙄